src/components/VExpansionPanel/VExpansionPanel.js

```diff
@@ -33,10 +32,38 @@ export default {
     }
   },
 
+  computed: {
+    classes () {
+      return {
+        'v-expansion-panel--focusable': this.focusable,
+        'v-expansion-panel--popout': this.popout,
+        'v-expansion-panel--inset': this.inset,
+        ...this.themeClasses
+      }
+    }
+  },
+
   watch: {
-    expand (v) {
-      this.open = Array(this.items.length).fill(false)
-      this.$emit('input', v ? this.open : null)
+    expand (val) {
+      let open
+      if (val) {
+        // Keep the currently open panel, there should already only be one
+      } else {
+        // Close all panels unless only one is open
+        const openCount = this.open.reduce((acc, val) => acc + val, 0)
+
+        if (openCount === 1) {
+          open = this.open.indexOf(true)
+        }
+
+        this.open = Array(this.items.length).fill(false)
```

Comment:
this should go into a variable and assigned after all computations. `this.open` is mutated twice in a row. This may not be an issue if it all runs on the same tick, but seems unnecessary either way.